### ISSUE TITLE
[Idea] Infinity enchant changes.

### ISSUE THREAD

**Issue by RDKRACZ** [ZoeyTheEgoist/Indexed#2]
Make infinity have levels. It's too powerful.
- Infinity I has a 20% chance to not use an arrow
Infinity II 40%; Infinity III 60%; Infinity IV 80%; Infinity V - vanilla behavior.

Make infinity work on crossbows.

**Comment by DDZiroWan** [comment]
I had come here just to make a extremely similar suggestion, instead i think infinity should behave somewhat like unbreaking, with a 1/(level+1) chance of consuming an arrow.

This would allow for using infinity with spectral and tipped arrows, it wouldn't make getting arrows useless, and would make infinity more balanced on crossbows.

Infinity I: 50% chance of consuming an arrow;
Infinity II: 33,3% chance of consuming an arrow;
Infinity III: 25% chance of consuming an arrow;
Infinity IV: 20% chance of consuming an arrow;
Infinity V: 16,7% chance of consuming an arrow;

**Comment by DDZiroWan** [comment]
I agree that infinity is overpowered and it makes arrows obsolete, no wonder many mods try to fix it by removing the need for arrows totally.
